nh_eobs_ldap/models/web_main.py

```diff
@@ -0,0 +1,68 @@
+from openerp import http
+from openerp.http import request
+from openerp.osv.osv import except_osv
+from openerp.addons.web.controllers.main import Session
+import openerp.modules as addons
+from openerp.tools.translate import _
+import operator
+
+
+class MainSession(Session):
+    """
+    A class to change the nh_eobs_api.controllers.routes method
+    """
+
+    @staticmethod
+    def _change_password(request, fields):
+        """
+        Actual implementation behind the change_password call
+
+        :param fields: submitted fields
+        :return: the new password or an error
+        """
+        old_password, new_password, confirm_password = \
+            operator.itemgetter('old_pwd', 'new_password', 'confirm_pwd')(
+                dict(map(operator.itemgetter('name', 'value'), fields)))
+        password_set = old_password.strip() \
+            and new_password.strip() \
+            and confirm_password.strip()
+        if not password_set:
+            return {'error': _('You cannot leave any password empty.'),
+                    'title': _('Change Password')}
+        if new_password != confirm_password:
+            return {
+                'error': _('The new password and its confirmation '
+                           'must be identical.'),
+                'title': _('Change Password')
+            }
+        try:
+            users_model = request.session.model('res.users')
+            user = users_model.browse(request.uid)
+            if user.change_password(old_password, new_password):
+                return {'new_password': new_password}
+        except Exception as e:
+            if isinstance(e, except_osv) \
+                    and 'Trust managed account' in e.value:
+                        return {'error': _(e.value),
+                                'title': _(e.name)}
+            return {'error': _(
+                'The old password you provided is incorrect, '
+                'your password was not changed.'),
+                'title': _('Change Password')}
```

Comment:
I like all the friendly error messages but am just curious: did the original method not provide messages like this?

Reply:
Looks like the old method provides those messages. Could we import it from `odoo_addons`, call it, and then just catch exceptions to add your trust messages?

Reply:
👍